openedx/features/content_tagging/models.py

```diff
@@ -0,0 +1,121 @@
+"""
+Content Tagging models
+"""
+from django.db import models
+from django.utils.translation import gettext as _
+from opaque_keys import InvalidKeyError, OpaqueKey
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from opaque_keys.edx.locator import LibraryLocatorV2, LibraryUsageLocatorV2
+from openedx_tagging.core.tagging.models import ObjectTag, Taxonomy
+from organizations.models import Organization
+
+
+class TaxonomyOrg(models.Model):
+    """
+    Represents the many-to-many relationship between Taxonomies and Organizations.
+
+    We keep this as a separate class from ContentTaxonomy so that class can remain a proxy for Taxonomy, keeping the
+    data models and usage simple.
+    """
+
+    class RelType(models.TextChoices):
+        OWNER = "OWN", _("owner")
+
+    taxonomy = models.ForeignKey(Taxonomy, on_delete=models.CASCADE)
+    org = models.ForeignKey(Organization, on_delete=models.CASCADE)
+    rel_type = models.CharField(
+        max_length=3,
+        choices=RelType.choices,
+        default=RelType.OWNER,
+    )
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["taxonomy", "rel_type"]),
+            models.Index(fields=["taxonomy", "rel_type", "org"]),
+        ]
+
+    @classmethod
+    def get_owners(
+        cls,
+        taxonomy: Taxonomy,
+    ) -> models.QuerySet:
+        """
+        Returns the org "owners" on the given taxonomy.
+
+        If the returned list is empty, then the returned taxonomy is enabled for all organizations.
+        """
+        return cls.objects.filter(
+            taxonomy=taxonomy,
+            rel_type=cls.RelType.OWNER,
+        )
+
+    @classmethod
+    def is_owner(cls, taxonomy: Taxonomy, org_short_name: str) -> bool:
+        """
+        Returns True if there's an "owner" relationship between the given taxonomy and an org with the given short name.
+
+        Or, if no "owner" relationships exist for the given taxonomy, then all orgs are enabled for that taxonomy.
+        """
+        org_owners = cls.get_owners(taxonomy)
+        if not org_owners.exists():
+            return True
```

Comment:
Is this function `is_enabled` or `is_owner`? Because saying "if nobody is the owner, everyone is the owner" seems weird.

Reply:
I agree that it's weird.. my only other idea was to add a field to ContentTaxonomy, e.g `available_to_all_orgs`, but I'd have to use a non-proxy model, and those won't cast (because they don't have all the fields).

I also don't want to call this `is_enabled`, because Taxonomy has an `enabled` field..  Do you have any better ideas?

Reply:
One problem with this current approach is that if an `Organization` is deleted, the `TaxonomyOrg` will be CASCADE deleted too, and then their taxonomy will suddenly go from private to "available for all orgs".

Maybe you can allow `org` to be null, so if there's a `TaxonomyOrg` with `org=None`, that means "make available to all orgs"? Otherwise you'd have to add a `extra_data` JSON field to the base Taxonomy class, which it's better to avoid for now if we can.

Reply:
Excellent point, thank you! Fixed with [03447f9](https://github.com/openedx/edx-platform/pull/32661/commits/03447f93fb593e1aa2001ffffccb3bf4dc41a215).